lib/badge.rb

```diff
@@ -0,0 +1,75 @@
+class Badge
+  include DSL
+
+  attr_reader :title, :description, :goal, :rank, :user
+
+  def self.slug
+    name.underscore.dasherize.sub('-badge', '')
+  end
+
+  def slug
+    self.class.slug
+  end
```

Comment:
These `slug` methods don't seem to be used for anything (instead you're using the class name, which makes more sense), so maybe just remove 'em